### PR TITLE
Mobt 157 nbhood refactor move plugins

### PR DESCRIPTION
**Pull request by fionaRust**
Partially addresses https://github.com/metoppv/mo-blue-team/issues/157

No functional changes, just moves all neighbourhood plugins into one file. Also updates the checksum after an acceptance test was removed in https://github.com/metoppv/improver/pull/1679

Testing:
 - [ ] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

### CODE REVIEW

**Review by bayliffe — APPROVED** [review]
Looks fine. Code coverage wants you to check a few lines are tested.

**Review by mspelman07 — APPROVED** [review]
I can't see any errors with this other than the Codecov error Ben mentioned.

File: improver/nbhood/nbhood.py
Comment (fionaRust) [inline improver/nbhood/nbhood.py]: I tried to find a case where the exception was raised so I could test it. It's not raised for a list of percentiles, a single percentile or the default percentiles. Maybe this was needed for an earlier version of python? Anyway as I couldn't test it I decided to remove it.

File: improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py
Comment (fionaRust) [inline improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py:389]: Added a new simple test for a single percentile. This was just to make sure it worked for this case and it seems fine so left it in.

File: improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py
Comment (fionaRust) [inline improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py:419]: I've modified this test to improve test coverage. We don't support multitime coordinates in improver, but you could have multiple realizations. This test now uses an input cube with two realizations, and creates an output cube with a leading dimension of realization followed by the percentile dimension.

File: improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py
Comment (fionaRust) [reply improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py:389]: Now this test returns a cube with a scalar percentile dimension. Also added a test for a single realization dimension

**Comment by bayliffe** [comment]
The pushed changes have not dismissed our now stale review approvals for some reason. Odd. I have checked the changes and I am happy with this PR.

**Comment by mspelman07** [comment]
I have also checked over the recent changes and I'm happy with them.

**Pull request merged**
